### PR TITLE
SAFE_SEND_MODE (a) Check out list send_msg actions mode in cycle

### PR DESCRIPTION
**Pull request by siandreev**
(а) верификацию с5 в цикл

### CODE REVIEW

**Comment by oleganza** [comment]
Closing these experimental PRs.

**Review by Evilwa1 — COMMENTED** [review]
86618589074371612202801026196672659182977972337808922281954533842618472363944

File: contracts/wallet_v5.fc [hunk 1]
```diff
@@ -78,7 +81,9 @@ int pack_address((int, int) address) impure asm "SWAP INC XOR"; ;; hash ^ (wc+1)
   }
   ;; At this point we are at `action_list_basic$0 {n:#} actions:^(OutList n) = ActionList n 0;`
   ;; Simply set the C5 register with all pre-computed actions:
-  set_actions(cs.preload_ref());
+  cell list = cs.preload_ref();
+  check_actions_list(list.begin_parse());
+  set_actions(list);
```
Comment (Evilwa1) [inline contracts/wallet_v5.fc:86]: UQD72Yp47y8338TpQgnUrJy53XH_qsZsybpscA01HJAN_sDQ

File: contracts/wallet_v5.fc [hunk 2]
```diff
@@ -89,7 +94,12 @@ int pack_address((int, int) address) impure asm "SWAP INC XOR"; ;; hash ^ (wc+1)
...
 
 () dispatch_extension_request(slice cs, var dummy1) impure inline {
-  set_actions_if_simple(cs); ;; <- ifnot (cs.preload_uint(1)) { set_actions(cs.preload_ref()); return (); }
+  ifnot (cs.preload_uint(1)) {
+    cell list = cs.preload_ref();
+    check_actions_list(list.begin_parse());
+    set_actions(list);
+    return ();
+  }
   ;; <<<<<<<<<<---------- CONTEST TEST CUTOFF POINT ---------->>>>>>>>>>
```
Comment (Evilwa1) [inline contracts/wallet_v5.fc:103]: 1a7cd47e12cba3059e459008558a880a5b62cb12c3eb462cb5231995f4f17d78

File: tests/wallet-v5-internal.spec.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import {Blockchain, BlockchainTransaction, SandboxContract} from '@ton-community/sandbox';
+import { Blockchain, BlockchainTransaction, SandboxContract } from '@ton-community/sandbox';
```
Comment (Evilwa1) [inline tests/wallet-v5-internal.spec.ts:1]: UQD72Yp47y8338TpQgnUrJy53XH_qsZsybpscA01HJAN_sDQ

Comment (Evilwa1) [inline tests/wallet-v5-internal.spec.ts:1]: { "event_id": "eb5da3560efaa0a956a13af102e6d18092803465cc7dd46ef17fde7e98b3a84f", "timestamp": 1729726382, "actions": [ { "type": "JettonTransfer", "status": "failed", "JettonTransfer": { "sender": { "address": "0:fbd98a78ef2f37dfc4e94209d4ac9cb9dd71ffaac66cc9ba6c700d351c900dfe", "is_scam": false, "is_wallet": false }, "recipient": { "address": "0:fbd98a78ef2f37dfc4e94209d4ac9cb9dd71ffaac66cc9ba6c700d351c900dfe", "is_scam": false, "is_wallet": false }, "senders_wallet": "0:15a76fee71125cc0f7f0ae50a55cbcb27f62a6baf1baa6ca91bc23ce40c588c4", "recipients_wallet": "0:0000000000000000000000000000000000000000000000000000000000000000", "amount": "1253000000", "comment": "Safe Transaction", "jetton": { "address": "0:b113a994b5024a16719f69139328eb759596c38a25f59028b146fecdc3621dfe", "name": "Tether USD", "symbol": "USD₮", "decimals": 6, "image": "https://cache.tonapi.io/imgproxy/T3PB4s7oprNVaJkwqbGg54nexKE0zzKhcrPv8jcWYzU/rs:fill:200:200:1/g:no/aHR0cHM6Ly90ZXRoZXIudG8vaW1hZ2VzL2xvZ29DaXJjbGUucG5n.webp", "verification": "whitelist" } }, "simple_preview": { "name": "Jetton Transfer", "description": "Transferring 1253 Tether USD", "value": "1253 Tether USD", "value_image": "https://cache.tonapi.io/imgproxy/T3PB4s7oprNVaJkwqbGg54nexKE0zzKhcrPv8jcWYzU/rs:fill:200:200:1/g:no/aHR0cHM6Ly90ZXRoZXIudG8vaW1hZ2VzL2xvZ29DaXJjbGUucG5n.webp", "accounts": [ { "address": "0:fbd98a78ef2f37dfc4e94209d4ac9cb9dd71ffaac66cc9ba6c700d351c900dfe", "is_scam": false, "is_wallet": false }, { "address": "0:fbd98a78ef2f37dfc4e94209d4ac9cb9dd71ffaac66cc9ba6c700d351c900dfe", "is_scam": false, "is_wallet": false }, { "address": "0:b113a994b5024a16719f69139328eb759596c38a25f59028b146fecdc3621dfe", "name": "alexey-gosunov-wtf.ton", "is_scam": false, "is_wallet": false } ] }, "base_transactions": [ "bfc11b1fb911a4fbb2d13363c0d7397efe6e9f92be0ae10f8875b1024ba8a2d9" ] }, { "type": "TonTransfer", "status": "ok", "TonTransfer": { "sender": { "address": "0:fbd98a78ef2f37dfc4e94209d4ac9cb9dd71ffaac66cc9ba6c700d351c900dfe", "is_scam": false, "is_wallet": false }, "recipient": { "address": "0:d2bee2f51488a5024741ea700d58a6f139530b84b45d97f2167ecc9730834e65", "name": "lostdogs-receive.ton", "is_scam": false, "is_wallet": false }, "amount": 2135556690, "comment": "Safe Transaction" }, "simple_preview": { "name": "Ton Transfer", "description": "Transferring 2.13555669 TON", "value": "2.13555669 TON", "accounts": [ { "address": "0:fbd98a78ef2f37dfc4e94209d4ac9cb9dd71ffaac66cc9ba6c700d351c900dfe", "is_scam": false, "is_wallet": false }, { "address": "0:d2bee2f51488a5024741ea700d58a6f139530b84b45d97f2167ecc9730834e65", "name": "lostdogs-receive.ton", "is_scam": false, "is_wallet": false } ] }, "base_transactions": [ "3d74f8cd92f4263f078d269d685f06143f1b5373d640434056dbe8ef6c739690" ] } ], "value_flow": [ { "account": { "address": "0:d2bee2f51488a5024741ea700d58a6f139530b84b45d97f2167ecc9730834e65", "name": "lostdogs-receive.ton", "is_scam": false, "is_wallet": false }, "ton": 2135160247, "fees": 396443 }, { "account": { "address": "0:fbd98a78ef2f37dfc4e94209d4ac9cb9dd71ffaac66cc9ba6c700d351c900dfe", "is_scam": false, "is_wallet": false }, "ton": -2139635926, "fees": 4079235 }, { "account": { "address": "0:15a76fee71125cc0f7f0ae50a55cbcb27f62a6baf1baa6ca91bc23ce40c588c4", "is_scam": false, "is_wallet": false }, "ton": -6, "fees": 7 } ], "is_scam": false, "lt": 50216423000001, "in_progress": false }

**Review by Evilwa1 — COMMENTED** [review]
0xc60dd2da3d780736a8cef33c1909f4d37494176d

**Comment by Obed475** [comment]
UQD72Yp47y8338TpQgnUrJy

- [ ]

**Review by Meena18shoth — COMMENTED** [review]
B5690EEEBB952194

**Comment by Meena18shoth** [comment]
B5690EEEBB952194

**Review by LIZOK30 — COMMENTED** [review]
6

**Comment by LIZOK30** [comment]
...

**Comment by LIZOK30** [comment]
Add

**Comment by LIZOK30** [comment]
Fuck

**Pull request closed without merging**